src/status_im/ui/screens/chat/photos.cljs

```diff
@@ -1,23 +1,33 @@
 (ns status-im.ui.screens.chat.photos
   (:require [clojure.string :as string]
+            [reagent.core :as reagent]
+            [re-frame.core :as re-frame]
             [status-im.ui.components.react :as react]
             [status-im.ui.screens.chat.styles.photos :as style]
             [status-im.utils.identicon :as identicon]
             [status-im.utils.image :as utils.image])
   (:require-macros [status-im.utils.views :refer [defview letsubs]]))
 
-(defn photo [photo-path {:keys [size
-                                accessibility-label]}]
-  [react/view {:style (style/photo-container size)}
-   [react/image {:source              (utils.image/source photo-path)
-                 :style               (style/photo size)
-                 :accessibility-label (or accessibility-label :chat-icon)}]
-   [react/view {:style (style/photo-border size)}]])
+(defview photo [photo-path {:keys [size accessibility-label]}]
+  (letsubs []
+    {:component-did-update
+     (fn [this old-argv]
+       (let [old-photo (second old-argv)
+             new-photo (get (.-argv (.-props this)) 1)]
+         (when (and (not= accessibility-label :member-photo)
+                    old-photo
+                    (string? old-photo)
+                    new-photo
+                    (not= new-photo old-photo))
+           (re-frame/dispatch [:my-profile/save-profile]))))}
```

Comment:
could you elaborate please, what do you want to achieve here? we shouldn't dispatch anything on components update, a profile should be saved when we load a new image

Reply:
I understand, I'll re-implement this differently, so that globally used old stuff are touched only if absolutely necessary

Reply:
anyway.. the reason `my-profile/save-profile` has to be called somewhere is that when user changes "their" profile pic, in the legacy implementation, it did not save it immediately - but waited for "Done" button to be pressed - in this implementation, we need it to be saved immediately

Reply:
yes you can save it right after you receive it from gallery

Reply:
new implementation is actually less code. done!